### DISCUSSION TITLE
Lilygo T7_v1.5 and Lolin TFT-2.4

### DISCUSSION THREAD

**Discussion by eboston** [HASwitchPlate/openHASP#245] [Hardware]
I got a Lilygo T7_v1.5 module and a Lolin TFT-2.4.  What firmware am I suppose to use and what do I need to do to the display to get it working?  I soldered a bridge for PWD control of the LED, but what about the other two connections?

**Comment by fvanroie** [comment]
The T7 has the same pinout as the `D1 mini ESP32`. The firmware for those boards is `d1-mini-esp32_ili9341_full_4MB_v0.6.2.bin`.

On the display, only solder `TFT-LED` to `D1`. The other jumpers are connected by default, no need to solder `TFT-CS` or `TFT-RST`.
Be sure to only solder a row of 1x8 male pins to the *inner* pads TXD->5V and RST->3V3 on each side of the ESP32 *(don't use double rows of pinheaders!)*

Fit the ESP32 onto the LCD, carefully checking the pins are in the correct socket. Then flash it and you should see the setup screen.

**Reply by eboston** [reply]
Thanks.  That is what I have done and have it working somewhat.  I can't seem to get the touch screen to calibrate correctly though.  At times, it acts like a ghost is touching it as it moves from one corner to the next by itself.

**Reply by fvanroie** [reply]
The ghost touches can be any number of things... configuration issue, electrical issue or even a mechanical issue...
If not done so already, remove the protective film from the touchscreen.

For configuration and checking the electric connections, please check [this post](https://github.com/HASwitchPlate/openHASP/discussions/131#discussioncomment-1450777).

To rule out a software issue, you can try loading another sketch to see if that one is also registering ghost touches.
Double and triple checking the pin configuration and looking for shorts/bad connections would be my first plan of attack.

**Reply by eboston** [reply]
Tried using a different display and still having issues with the touch.  I was checking the pin assignment in the d1-mini-esp32_ili9341.ini file and it shows TOUCH_CS goes to D17.  This pin is not exposed on the D1 Mini ESP32 that I can tell.  Checking the TFT-2.4 board, I found the TS_CS connects to D27 (marked D3 on the TFT-2.4 board).  Am I not reading the file correctly?  Not having the CS wired correctly could be causing a problem.

**Reply by fvanroie** [reply]
hmmm, it seems the pins _have_ changed from T7 1.4 to T7 1.5 😞 
`17` should be `27` in your case... but it requires recompiling from source.
Are you able to try that? If not I can make a test build...

**Reply by eboston** [reply]
I was able to get the firmware to compile and uploaded it.  Still had the same problem.  I ordered two of the displays so connected to that one and it worked.  Moved back to the first display and no touch response.  Looks like it is a bad display that is causing the problems.

**Reply by fvanroie** [reply]
Having access to multiple devices helps to find hardware issues. Good to hear you got it to work!

**Comment by eboston** [comment]
I have been trying to compile but have been having trouble.  Just haven't
had time to work the problems out.
